### PR TITLE
[DDW-53] Enable "Wallet recovery phrase verification" feature

### PR DESCRIPTION
**Pull request by daniloprates**
This PR integrates the `cardano-js` method for verifying a wallet's recovery phrase.

Obs: It also includes **[DDW-33] Integrate recovery phrase verification JS package**.

## Todos:

- [x] Re-enable "Recovery phrase verification" feature
- [x] Accept 12 and 15-word recovery phrase wallets
  - [x] Correct 12 and 15-word copy and validations for the wallet being checked
- [x] Reset verification form on dialog open
- [x] Replace hardcoded number of words with CONST
- [x] Separate the local wallet data handling from the recovery phrase verification feature 
- [x] Improve types
- [x] Fix [tests](https://github.com/input-output-hk/daedalus/blob/feature/ddw-33-integrate-recovery-phrase-verification-js-package/tests/nodes/e2e/features/wallet-settings-recovery-phrase-verification.feature)
- [x] Improve the "time-ago" and "time-to" logic in the widget
- [x] Fix `cardano-js` import error

## Screenshots:

Screenshot 2020-05-04 at 13 01 06
Screenshot 2020-05-04 at 13 01 13

### CODE REVIEW

**Comment by DmitriiGaico** [comment]
@daniloprates Hello Danilo. I have added a point in "To Do" section as currently Mnemonic Input is not purged and once you open Verification Window it shows you a pre-entered Mnemonic. 
Mnemonic

File: source/renderer/app/containers/wallet/dialogs/settings/WalletRecoveryPhraseContainer.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      recoveryPhraseCheck,
+      recoveryPhraseClose,
+    } = actions.walletBackup;
+    const { openExternalLink } = stores.app;
+    const onContinue =
+      this.step === 2
+        ? recoveryPhraseCheck.trigger
+        : recoveryPhraseContinue.trigger;
+    const onClose = recoveryPhraseClose.trigger;
+    const wordCount = activeWallet.discovery === 'random' ? 12 : 15;
```
Comment (nikolaglumac) [inline source/renderer/app/containers/wallet/dialogs/settings/WalletRecoveryPhraseContainer.js:46]: @daniloprates please replace 12/15 with constants 🙏

File: source/renderer/app/containers/wallet/WalletSettingsPage.js [hunk 2]
```diff
@@ -62,65 +60,62 @@ export default class WalletSettingsPage extends Component<Props> {
       recoveryPhraseVerificationStatusType,
     } = getWalletRecoveryPhraseVerification(activeWallet.id);
 
+    const wordCount = activeWallet.discovery === 'random' ? 12 : 15;
```
Comment (nikolaglumac) [inline source/renderer/app/containers/wallet/WalletSettingsPage.js:63]: @daniloprates please replace 12/15 with constants 🙏

**Comment by ManusMcCole** [comment]
On MacOS Daedalus Selfnode and Flight both show this error message when I try to launch. 
image

**Comment by ManusMcCole** [comment]
There is a small issue in Storybook only in Japanese where the English words for how long since the wallet was verified do not get translated 
image

**Comment by nikolaglumac** [comment]
@tomothespian please review this one 🙏

**Comment by gnpf** [comment]
@daniloprates This is enabled for paper wallets and only allows 12 words for verification

**Comment by nikolaglumac** [comment]
> @daniloprates This is enabled for paper wallets and only allows 12 words for verification
> 
> 

@gnpf this is a known limitation. We will not be fixing this in this PR. 
However, we might add one sentence which will explain to the user that if his wallet was restored from the paper wallet certificate the verification won't work.

**Comment by ManusMcCole** [comment]
Still seems to be a small issue with Storybook for wallets verified 2 months ago 
image

**Comment by ManusMcCole** [comment]
Also for one month ago 
image

**Comment by daniloprates** [comment]
@ManusMcCole they are both correct as these are the times remaining to 6 months.

File: source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+      message,
+      timeAgo,
+      timeUntilWarning,
+    } = this.recoveryPhraseStatus;
+    const { recoveryPhraseVerificationStatus } = getStatusFromWalletData({
+      creationDate,
+      recoveryPhraseVerificationDate,
+    });
+
+    const statusStyle = camelCase(`status ${recoveryPhraseVerificationStatus}`);
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/wallet/settings/WalletRecoveryPhraseVerificationWidget.js:206]: It won't work. If you check [this gist](https://jsbin.com/rukuhob/edit?js,console), only the first one works fine.

File: source/renderer/app/components/navigation/NavDropdown.scss [hunk 4]
```diff
@@ -7,6 +7,24 @@
     width: 100%;
   }
 
+  .optionLabel {
+    &.hasNotification {
+      // :global {
+      // .SimpleBubble_bubble {
```
[the remark this answers is not in the document]
Comment (daniloprates) [reply source/renderer/app/components/navigation/NavDropdown.scss]: ✅

**Comment by nikolaglumac** [comment]
This one can be merged as soon as we add missing Japanese translation: https://github.com/input-output-hk/daedalus/pull/1929/files#diff-040f484ab749aa74c4c2dd5afcdf3986R745

**Pull request merged**